src/javascript/app/Modules/Reports/Helpers/shortcode.js

```diff
@@ -1,27 +1,28 @@
 const Shortcode = (() => {
-    let info_from_shortcode = {};
+    const info_from_shortcode = {
+        category  : '',
+        underlying: '',
+        barrier_1 : '',
+    };
 
     const extractInfoFromShortcode = (shortcode) => {
-        const pattern = new RegExp('^([A-Z]+)_((OTC_[A-Z0-9]+)|R_[\\d]{2,3}|[A-Z]+)_([\\d.\\d]+)_(\\d+\\w)_(\\d+\\w)_(S[\\d.\\d]+P|(?:\\d+))_(\\d+)'); // Used to get market name from shortcode
+        const pattern = new RegExp('^([A-Z]+)_((OTC_[A-Z0-9]+)|R_[\\d]{2,3}|[A-Z]+)'); // Used to get market name from shortcode
         const extracted = pattern.exec(shortcode);
         if (extracted !== null) {
             info_from_shortcode.category   = extracted[1].toLowerCase();
             info_from_shortcode.underlying = extracted[2];
-            info_from_shortcode.atm        = extracted[7];
+
+            if (/^(CALL|PUT)$/i.test(info_from_shortcode.category)) {
+                info_from_shortcode.barrier_1 = shortcode.split('_').slice(-2)[0];
+            }
         }
 
         return info_from_shortcode;
     };
 
     const isHighLow = (shortcode) => {
-        if (shortcode) {
-            info_from_shortcode = extractInfoFromShortcode(shortcode);
-        }
-        if (/CALL|PUT/i.test(info_from_shortcode.category)) {
-            return !/^S0P$/.test(info_from_shortcode.atm);
-        }
-
-        return false;
+        if (shortcode) extractInfoFromShortcode(shortcode);
+        return /^(CALL|PUT)$/i.test(info_from_shortcode.category) ? !/^S0P$/.test(info_from_shortcode.barrier_1) : false;
```

Comment:
same here, also better just check if `info_from_shortcode.barrier_1` instead of again checking the category